src/components/TodoItem.tsx

```diff
@@ -1,23 +1,35 @@
-import { ChangeEvent } from "react";
+import { ChangeEvent, MouseEvent } from "react";
 import { TodoItemType } from "../App";
 
 type TodoItemProps = {
     index: number;
     item: TodoItemType;
-    handleTodoItemChecked: (index: number, newCheckedState: boolean) => void;
+    onTodoItemChecked: (index: number, newCheckedState: boolean) => void;
+    onDeleteClick: (index: number) => void;
 }
 
-export function TodoItem({ index, item, handleTodoItemChecked }: TodoItemProps) {
+export function TodoItem({ index, item, onTodoItemChecked, onDeleteClick }: TodoItemProps) {
     const { isDone, name } = item;
 
     const handleOnChange = (event: ChangeEvent<HTMLInputElement>) => {
       const newCheckedState = event.target.checked;
 
-      handleTodoItemChecked(index, newCheckedState);
+      onTodoItemChecked(index, newCheckedState);
     };
 
-    const handleDeleteClick = () => {
-      // TODO: build me!
+    function handleDeleteClick() {
+      onDeleteClick(index);
+    }
+
+    if (isDone){
```

Comment:
The code in these two blocks has a lot of duplication. We should clean it up to reduce it.

We can do this by extracting the `<p>{name}</p>` and `<p><s>{name}</s></p>` portion of these conditional renders into it's own React component.

```jsx
function TodoItemName({name, isDone}: { name: string; isDone: boolean; }) {
    if (!isDone) {
        return (<p>{name}</p>);
    }
    
    return (<p><s>{name}</s></p>);
}
``` 

Then, we can use that component in our main return below, and get rid of this condition here!

```jsx
return (
        <div style={{
          marginBottom: `2px`
        }}>
          <input type="checkbox" checked={isDone} onChange={handleOnChange}/>
          <TodoListItemName name={name} isDone={isDone} />
          <button onClick={handleDeleteClick}>Delete</button>
        </div>
    );
```

Reply:
handled in [77230de](https://github.com/ZachWitzig/todoist/pull/3/commits/77230de0aa48039c04a2d2707327e22d807c5848)